pyomo/contrib/interior_point/interface.py

```diff
@@ -0,0 +1,624 @@
+                            self._barrier / (self._nlp.get_primals() - self._nlp.primals_lb()) +
+                            self._barrier / (self._nlp.primals_ub() - self._nlp.get_primals()))
+        timer.stop('grad_lag_primals')
+
+        timer.start('grad_lag_slacks')
+        grad_lag_slacks = (-self._nlp.get_duals_ineq() -
+                           self._barrier / (self._slacks - self._nlp.ineq_lb()) +
+                           self._barrier / (self._nlp.ineq_ub() - self._slacks))
+        timer.stop('grad_lag_slacks')
+
+        rhs = BlockVector(4)
+        rhs.set_block(0, grad_lag_primals)
+        rhs.set_block(1, grad_lag_slacks)
+        rhs.set_block(2, eq_resid)
+        rhs.set_block(3, ineq_resid)
+        rhs = -rhs
+        return rhs
+
+    def set_primal_dual_kkt_solution(self, sol):
+        self._delta_primals = sol.get_block(0)
+        self._delta_slacks = sol.get_block(1)
+        self._delta_duals_eq = sol.get_block(2)
+        self._delta_duals_ineq = sol.get_block(3)
+
+    def get_delta_primals(self):
+        return self._delta_primals
+
+    def get_delta_slacks(self):
+        return self._delta_slacks
+
+    def get_delta_duals_eq(self):
+        return self._delta_duals_eq
+
+    def get_delta_duals_ineq(self):
+        return self._delta_duals_ineq
+
+    def get_delta_duals_primals_lb(self):
+        res = (((self._barrier - self._duals_primals_lb * self._delta_primals) /
+                (self._nlp.get_primals() - self._nlp.primals_lb())) -
+               self._duals_primals_lb)
+        return res
+
+    def get_delta_duals_primals_ub(self):
+        res = (((self._barrier + self._duals_primals_ub * self._delta_primals) /
+                (self._nlp.primals_ub() - self._nlp.get_primals())) -
+               self._duals_primals_ub)
+        return res
+
+    def get_delta_duals_slacks_lb(self):
+        res = (((self._barrier - self._duals_slacks_lb * self._delta_slacks) /
+                (self._slacks - self._nlp.ineq_lb())) -
+               self._duals_slacks_lb)
+        return res
+
+    def get_delta_duals_slacks_ub(self):
+        res = (((self._barrier + self._duals_slacks_ub * self._delta_slacks) /
+                (self._nlp.ineq_ub() - self._slacks)) -
+               self._duals_slacks_ub)
+        return res
+
+    def evaluate_objective(self):
+        return self._nlp.evaluate_objective()
+
+    def evaluate_eq_constraints(self):
+        return self._nlp.evaluate_eq_constraints()
+
+    def evaluate_ineq_constraints(self):
+        return self._nlp.evaluate_ineq_constraints()
+
+    def evaluate_grad_objective(self):
+        return self._nlp.evaluate_grad_objective()
+
+    def evaluate_jacobian_eq(self):
+        return self._nlp.evaluate_jacobian_eq()
+
+    def evaluate_jacobian_ineq(self):
+        return self._nlp.evaluate_jacobian_ineq()
+
+    def regularize_equality_gradient(self, kkt, coef, copy_kkt=True):
+        # Not technically regularizing the equality gradient ...
```

Comment:
Hmmm. I agree with your comment. Maybe we should change the name?

Reply:
We could have a single method named `regularize_kkt(self, kkt, hess_coef, grad_coef, copy_kkt=True)`?

Reply:
I can't think of better names if we keep the methods split apart.

Reply:
I like regularize_kkt. hess_coef and eqcon_coef?

Reply:
What about hess_coefficients and eqcon_coefficients in a vector format instead of a scalar. Even a block vector format in case the regularization wants to target a particular subproblem.